akka-http-cluster-management/src/main/scala/akka/cluster/http/management/ClusterHttpManagement.scala

```diff
@@ -0,0 +1,236 @@
+            complete(ClusterHttpManagementMessage(s"Leaving ${member.uniqueAddress.address}"))
+          case _ ⇒
+            complete(StatusCodes.BadRequest → ClusterHttpManagementMessage("Operation not supported"))
+        }
+      }
+    }
+
+  private def routesMember(cluster: Cluster) =
+    path(Remaining) { memberAddress ⇒
+      cluster.readView.members.find(m ⇒ s"${m.uniqueAddress.address}" == memberAddress) match {
+        case Some(member) ⇒
+          routeGetMember(cluster, member) ~ routeDeleteMember(cluster, member) ~ routePutMember(cluster, member)
+        case None ⇒
+          complete(StatusCodes.NotFound → ClusterHttpManagementMessage(s"Member [$memberAddress] not found"))
+      }
+    }
+
+  def apply(cluster: Cluster): Route = apply(cluster, "members")
+
+  def apply(cluster: Cluster, pathPrefixName: String): Route =
+    pathPrefix(pathPrefixName) {
+      pathEndOrSingleSlash {
+        routeGetMembers(cluster) ~ routePostMembers(cluster)
+      } ~
+        routesMember(cluster)
+    }
+
+  def apply(cluster: Cluster, asyncAuthenticator: AsyncAuthenticator[String]): Route = {
+    authenticateBasicAsync[String](realm = "secured", asyncAuthenticator) { _ ⇒
+      apply(cluster)
+    }
+  }
+
+  def apply(cluster: Cluster, pathPrefixName: String, asyncAuthenticator: AsyncAuthenticator[String]): Route = {
+    authenticateBasicAsync[String](realm = "secured", asyncAuthenticator) { _ ⇒
+      apply(cluster, pathPrefixName)
+    }
+  }
+}
+
+object ClusterHttpManagement {
+  def apply(cluster: Cluster): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster)
+
+  def apply(cluster: Cluster, pathPrefix: String): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), None, None)
+
+  def apply(cluster: Cluster, asyncAuthenticator: AsyncAuthenticator[String]): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, None, Some(asyncAuthenticator), None)
+
+  def apply(cluster: Cluster, https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, None, None, Some(https))
+
+  def apply(cluster: Cluster, pathPrefix: String, asyncAuthenticator: AsyncAuthenticator[String]): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), Some(asyncAuthenticator), None)
+
+  def apply(cluster: Cluster, pathPrefix: String, https: ConnectionContext) =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), None, Some(https))
+
+  def apply(cluster: Cluster, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, None, Some(asyncAuthenticator), Some(https))
+
+  def apply(cluster: Cluster, pathPrefix: String, asyncAuthenticator: AsyncAuthenticator[String], https: ConnectionContext): ClusterHttpManagement =
+    new ClusterHttpManagement(cluster, Some(pathPrefix), Some(asyncAuthenticator), Some(https))
+
+  def create(cluster: Cluster): ClusterHttpManagement =
```

Comment:
good, thanks for all java api - it could just delegate to it's sibling apply I think